src/Microsoft.ML.Transforms/WhiteningTransform.cs

```diff
@@ -108,72 +115,95 @@ public bool TryUnparse(StringBuilder sb)
             }
         }
 
-        public sealed class ColInfoEx
+        public sealed class ColInfo
         {
+            public readonly string Input;
+            public readonly string Output;
             public readonly WhiteningKind Kind;
-            public readonly Float Epsilon;
+            public readonly float Epsilon;
             public readonly int MaxRow;
             public readonly bool SaveInv;
             public readonly int PcaNum;
-            public readonly VectorType Type;
 
-            public ColInfoEx(Column item, Arguments args, ColInfo info)
+            /// <summary>
+            /// Describes how the transformer handles one input-output column pair.
+            /// </summary>
+            /// <param name="input">Name of the input column.</param>
+            /// <param name="output">Name of the column resulting from the transformation of <paramref name="input"/>. Null means <paramref name="input"/> is replaced. </param>
+            /// <param name="kind">Whitening kind (PCA/ZCA).</param>
+            /// <param name="eps">Whitening constant, prevents division by zero.</param>
+            /// <param name="maxRows">Maximum number of rows used to train the transform.</param>
+            /// <param name="saveInverse">Whether to save inverse (recovery) matrix.</param>
+            /// <param name="pcaNum">In case of PCA whitening, indicates the number of components to retain.</param>
+            public ColInfo(string input, string output = null, WhiteningKind kind = Defaults.Kind, float eps = Defaults.Eps,
+                int maxRows = Defaults.MaxRows, bool saveInverse = Defaults.SaveInverse, int pcaNum = Defaults.PcaNum)
             {
+                Input = input;
+                Contracts.CheckValue(Input, nameof(Input));
+                Output = output ?? input;
+                Contracts.CheckValue(Output, nameof(Output));
+                Kind = kind;
+                Contracts.CheckUserArg(Kind == WhiteningKind.Pca || Kind == WhiteningKind.Zca, nameof(Kind));
+                Epsilon = eps;
+                Contracts.CheckUserArg(0 <= Epsilon && Epsilon < float.PositiveInfinity, nameof(Epsilon));
+                MaxRow = maxRows;
+                Contracts.CheckUserArg(MaxRow > 0, nameof(MaxRow));
+                SaveInv = saveInverse;
+                PcaNum = pcaNum; // REVIEW: make it work with pcaNum == 1.
+                Contracts.CheckUserArg(PcaNum >= 0, nameof(PcaNum));
+            }
+
+            internal ColInfo(Column item, Arguments args)
+            {
+                Input = item.Source;
+                Contracts.CheckValue(Input, nameof(Input));
+                Output = item.Name ?? item.Source;
```

Comment:
>item.Name ?? item.Source [](start = 25, length = 24)

in all our estimators we have following pattern:
`new ColumnInfo(item.Source ?? item.Name, item.Name)`
you have it other way around